src/Components/TableItem.js

```diff
@@ -314,18 +314,62 @@ TableItem.defaultProps = {
     status: 'Open',
     update: false,
   },
-  onDelete: (id) => {
-    const { tableItems } = this.state;
-    const index = tableItems.findIndex(x => x.id === id);
-    tableItems.splice(index, 1);
+  saveItem: item => ({
+    type: 'SAVE_ITEM',
+    item,
+  }),
+  deleteItem: id => ({
+    type: 'DELETE_ITEM',
+    id,
+  }),
+};
+
+
+NormalContent.propTypes = {
+  savedItem: PropTypes.shape({
+    id: PropTypes.string,
+    subject: PropTypes.string,
+    category: PropTypes.string,
+    assignee: PropTypes.string,
+    priority: PropTypes.string,
+    status: PropTypes.string,
+    update: PropTypes.bool,
+  }),
+  priorityClass: PropTypes.string,
+  statusClass: PropTypes.string,
+  editItem: PropTypes.func,
+  handleDeleteItem: PropTypes.func,
+};
+
+NormalContent.defaultProps = {
+  savedItem: {
+    id: 'Hyq2-P3GQ',
+    subject: 'A new rating has been received',
+    category: 'Marketing',
+    assignee: 'Erwin',
+    priority: 'Medium',
+    status: 'Open',
+    update: false,
+  },
+  priorityClass: 'badge badge-pill badge-secondary',
+  statusClass: 'badge badge-pill badge-primary',
+  editItem: () => {
+    const { tableItem } = this.props;
     this.setState({
-      tableItems,
+      savedItem: tableItem,
+      isEditing: true,
     });
   },
-  saveItem: (item) => {
-    const { onSave } = this.props;
-    onSave(item);
+  handleDeleteItem: () => {
+    const { tableItem, deleteItem } = this.props;
+    const theId = tableItem.id;
+    deleteItem(theId);
   },
 };
 
-export default TableItem;
+const mapDispatchToProps = dispatch => ({
+  deleteItem: bindActionCreators(deleteItem, dispatch),
+  saveItem: bindActionCreators(saveItem, dispatch),
+});
+
+export default connect(null, mapDispatchToProps)(TableItem);
```

Comment:
You need a container component for each presentational components when you're using `react-redux`
ref: https://redux.js.org/basics/usage-with-react#implementing-container-components

Reply:
Highly recommended to follow the file structure like [Redux TodoMVC Example](https://github.com/reduxjs/redux/blob/master/examples/todomvc/README.md)